## Part 1: Explore the Data

Import the data and use Pandas to learn more about the dataset.

In [821]:
import pandas as pd
pd.set_option('display.max_rows', 500)
df = pd.read_csv('Resources/client_dataset.csv')
df.head(10)

first      last                                           job  \
0     Donald   Harding                                  Immunologist   
1    Tiffany     Myers                               Music therapist   
2    Shannon    Watson                                  Immunologist   
3     Nathan     Baker                         Accounting technician   
4  Christina  Schwartz                                  Chiropractor   
5    Jessica     Reyes  Production designer, theatre/television/film   
6      Brian     Moore                                       Barista   
7      Jesus    Wright                            Theme park manager   
8    Rebecca     Brown                                   Chiropodist   
9       Greg    Garcia                            Secretary, company   

                 phone                                email  client_id  \
0   793-904-7725x39308     harding.donald.7185@sullivan.com      58515   
1     201.442.4543x942    myers.t.6537@ferguson-johnson.net      37609   
2    687.737.9424x8503                swatson8146@payne.net      57113   
3     827-788-8123x012               bakernathan@benson.com      46554   
4         265-829-3643  christinaschwartz9252@mcconnell.com      92089   
5           5016915923             jessica.reyes@grimes.net      33615   
6  +1-893-400-2733x982          moore.official@martinez.com      87213   
7   605.947.9467x66513        wright.jesus.7579@meadows.com      46709   
8     665-955-0156x926                 rbrown850@lewis.info      73362   
9   (231)342-7216x1421                  ggarcia@bentley.biz      58789   

   order_id  order_date  order_week  order_year         item_id     category  \
0   8953482  2023-04-28          17        2023  EUD29711-63-6U        decor   
1   8069089  2023-05-19          20        2023  XDA18116-89-4A  consumables   
2   1902144  2023-01-29           4        2023  ABE59463-05-7E     software   
3   9031802  2023-04-25          17        2023  ZMM00836-65-0C  consumables   
4   1322274  2023-05-28          21        2023  BZX55559-12-3X  consumables   
5   1327730  2023-05-02          18        2023  BMB23158-88-9A        decor   
6   8080280  2023-04-19          16        2023  UAC50632-25-4E  consumables   
7   8946782  2023-02-14           7        2023  UEU25316-01-1U  consumables   
8   9394118  2023-05-13          19        2023  DDA06248-98-7Z  consumables   
9   9665852  2023-04-14          15        2023  DUU85042-87-5E    furniture   

          subcategory  unit_price  unit_cost  unit_weight  qty  line_number  
0            wall art     1096.80     762.71         7.50  105            1  
1                pens       24.95      15.09         1.49   21            0  
2  project management       13.52       7.86         1.68   39            6  
3                pens       36.42      24.85         1.23   29            3  
4                misc      195.10     108.17        46.43   20            1  
5              plants      172.84     154.44         2.79   78            0  
6    kitchen supplies      245.56     231.95         1.73   56            5  
7                pens       20.99      16.97         5.12   22            4  
8      paper products       83.86      59.00         0.27   49            3  
9             seating       66.88      54.15         2.47   78            1

In [822]:
df.tail()

first       last                      job               phone  \
54634   Lauren      Reese           Radio producer  977.877.5272x11382   
54635  Derrick      Moore                 Musician        358.661.5483   
54636   Monica  Gutierrez         Graphic designer    294.805.9100x339   
54637    Wanda    Solomon             Toxicologist       (311)767-4924   
54638   Dwayne       Bell  Chief Financial Officer   606-619-7667x9508   

                                    email  client_id  order_id  order_date  \
54634  late_reese_4081@montoya-chavez.org      29043   9021716  2023-04-26   
54635        derrick.moore.2602@pope.info      41908   6290153  2023-02-08   
54636           gutierrezm3195@morris.org      35176   8692622  2023-03-05   
54637           solomonwanda5962@ross.org      24485   7592730  2023-01-18   
54638                    belld@galvan.com      24007   7489403  2023-01-04   

       order_week  order_year         item_id     category  \
54634          17        2023  UCA28532-31-0B     software   
54635           6        2023  EBD29277-23-7U  consumables   
54636           9        2023  EBD30093-55-9Z  consumables   
54637           3        2023  XMZ38239-49-3D    furniture   
54638           1        2023  ZCZ05589-50-5D     software   

              subcategory  unit_price  unit_cost  unit_weight   qty  \
54634           logistics       83.13      51.60         2.25    33   
54635    kitchen supplies      206.59     175.46        11.70    47   
54636   bathroom supplies       65.66      57.31         4.16   475   
54637              stands        1.48       1.22        18.04   112   
54638  project management        3.01       1.61         2.07  1031   

       line_number  
54634            8  
54635            0  
54636            6  
54637            0  
54638            5

In [823]:
# View the column names in the data
df.columns

Index(['first', 'last', 'job', 'phone', 'email', 'client_id', 'order_id',
       'order_date', 'order_week', 'order_year', 'item_id', 'category',
       'subcategory', 'unit_price', 'unit_cost', 'unit_weight', 'qty',
       'line_number'],
      dtype='object')

In [824]:
# Use the describe function to gather some basic statistics
df.describe(include='all')

first     last        job       phone                     email  \
count         54639    54639      54639       54639                     54639   
unique          360      477        498         999                       999   
top     Christopher  Johnson  Osteopath  5016915923  jessica.reyes@grimes.net   
freq           1090     1164        631         220                       220   
mean            NaN      NaN        NaN         NaN                       NaN   
std             NaN      NaN        NaN         NaN                       NaN   
min             NaN      NaN        NaN         NaN                       NaN   
25%             NaN      NaN        NaN         NaN                       NaN   
50%             NaN      NaN        NaN         NaN                       NaN   
75%             NaN      NaN        NaN         NaN                       NaN   
max             NaN      NaN        NaN         NaN                       NaN   

           client_id      order_id  order_date    order_week    order_year  \
count   54639.000000  5.463900e+04       54639  54639.000000  54639.000000   
unique           NaN           NaN         150           NaN           NaN   
top              NaN           NaN  2023-04-19           NaN           NaN   
freq             NaN           NaN         538           NaN           NaN   
mean    54837.869416  5.470190e+06         NaN     11.359139   2022.993064   
std     25487.438231  2.599807e+06         NaN      7.023499      0.082997   
min     10033.000000  1.000886e+06         NaN      1.000000   2022.000000   
25%     33593.000000  3.196372e+06         NaN      6.000000   2023.000000   
50%     53305.000000  5.496966e+06         NaN     11.000000   2023.000000   
75%     78498.000000  7.733869e+06         NaN     17.000000   2023.000000   
max     99984.000000  9.998480e+06         NaN     52.000000   2023.000000   

               item_id     category        subcategory    unit_price  \
count            54639        54639              54639  54639.000000   
unique             500            5                 18           NaN   
top     MAZ00314-80-8M  consumables  bathroom supplies           NaN   
freq               148        23538               6424           NaN   
mean               NaN          NaN                NaN    136.267207   
std                NaN          NaN                NaN    183.873135   
min                NaN          NaN                NaN      0.010000   
25%                NaN          NaN                NaN     20.800000   
50%                NaN          NaN                NaN     68.310000   
75%                NaN          NaN                NaN    173.160000   
max                NaN          NaN                NaN   1396.230000   

           unit_cost   unit_weight           qty   line_number  
count   54639.000000  54639.000000  5.463900e+04  54639.000000  
unique           NaN           NaN           NaN           NaN  
top              NaN           NaN           NaN           NaN  
freq             NaN           NaN           NaN           NaN  
mean       99.446073      5.004116  5.702646e+02      2.979667  
std       133.164267      5.326599  1.879552e+04      2.436320  
min         0.010000      0.000000  0.000000e+00      0.000000  
25%        14.840000      1.450000  3.200000e+01      1.000000  
50%        49.890000      3.240000  6.800000e+01      3.000000  
75%       125.570000      6.890000  1.700000e+02      5.000000  
max       846.270000     46.430000  3.958244e+06      9.000000

In [825]:
# Use the describe function to gather some basic statistics
df['first'].describe()

count           54639
unique            360
top       Christopher
freq             1090
Name: first, dtype: object

In [826]:
# Checking if df has any NaN values
df.isna().sum()

first          0
last           0
job            0
phone          0
email          0
client_id      0
order_id       0
order_date     0
order_week     0
order_year     0
item_id        0
category       0
subcategory    0
unit_price     0
unit_cost      0
unit_weight    0
qty            0
line_number    0
dtype: int64

In [827]:
# Checking if df has any NaN values
df.isnull().sum()

first          0
last           0
job            0
phone          0
email          0
client_id      0
order_id       0
order_date     0
order_week     0
order_year     0
item_id        0
category       0
subcategory    0
unit_price     0
unit_cost      0
unit_weight    0
qty            0
line_number    0
dtype: int64

In [828]:
# Use this space to do any additional research
# and familiarize yourself with the data.
# get types
df.dtypes

first           object
last            object
job             object
phone           object
email           object
client_id        int64
order_id         int64
order_date      object
order_week       int64
order_year       int64
item_id         object
category        object
subcategory     object
unit_price     float64
unit_cost      float64
unit_weight    float64
qty              int64
line_number      int64
dtype: object

In [829]:
# Use this space to do any additional research
# and familiarize yourself with the data.
# get shape
df.shape

(54639, 18)

In [830]:
# Use this space to do any additional research
# and familiarize yourself with the data.
# getting counts off some names 
df.head()
df['first'].value_counts()

first
Christopher    1090
Michael        1047
William         982
John            959
Jennifer        902
Daniel          791
Thomas          772
David           742
Jessica         730
Michelle        656
Brian           602
Melissa         598
Ashley          586
James           571
Robert          556
Mary            551
Richard         548
Brandon         526
Mark            522
Nicholas        502
Lisa            499
Lori            485
Nicole          467
Patrick         456
Sarah           453
Katie           404
Rebecca         403
Angela          395
Bryan           392
Julie           389
Anthony         389
Susan           382
Sara            355
Holly           346
Wendy           341
Sean            339
Peter           338
Alexandra       332
Joseph          329
Matthew         328
Danny           324
Doris           321
Elizabeth       320
Charles         316
Heather         313
Michele         309
Kelly           306
Tiffany         302
Barry           298
Samantha      

In [831]:
# Use this space to do any additional research
# and familiarize yourself with the data.
# get emails of the first 3 rows

df.iloc[0:3,4]

0     harding.donald.7185@sullivan.com
1    myers.t.6537@ferguson-johnson.net
2                swatson8146@payne.net
Name: email, dtype: object

In [832]:
# What three item categories had the most entries?
# Select the top three categories
# option 1
category_counts = df['category'].value_counts()
top_three_categories = category_counts.head(3)
top_three_categories

category
consumables    23538
furniture      11915
software        8400
Name: count, dtype: int64

In [833]:
# For the category with the most entries,
# which subcategory had the most entries?
# Select the top category
top_category = df['category'].value_counts().idxmax()
# Filter the DataFrame to include only the rows with the top category
top_category_df = df[df['category'] == top_category]
# Use value_counts() on the subcategory column within the filtered DataFrame
subcategory_counts = top_category_df['subcategory'].value_counts()
top_subcategory = subcategory_counts.head(1)
top_subcategory

subcategory
bathroom supplies    6424
Name: count, dtype: int64

In [834]:
# Which five clients had the most entries in the data?
top_clients_counts = df['client_id'].value_counts()
top_five_clients = top_clients_counts.head(5)
print(top_five_clients)

client_id
33615    220
66037    211
46820    209
38378    207
24741    207
Name: count, dtype: int64


In [835]:
# Store the client ids of those top 5 clients in a list.
cliants_as_list = list(top_five_clients.to_dict().keys())
print(cliants_as_list)

[33615, 66037, 46820, 38378, 24741]


In [836]:
# How many total units (the qty column) did the
# client with the most entries order order?

top_client = df['client_id'].value_counts().idxmax()
print(top_client)
top_client_df = df[df['client_id'] == top_client]
top_client_df['qty'].sum()

33615


64313

## Part 2: Transform the Data
Do we know that this client spent the more money than client 66037? If not, how would we find out? Transform the data using the steps below to prepare it for analysis.

In [837]:
# Create a column that calculates the 
# subtotal for each line using the unit_price
# and the qty
updated_df = df.copy()
updated_df['subtotal'] = updated_df['unit_price'] * updated_df['qty']
updated_df.head()


first      last                    job               phone  \
0     Donald   Harding           Immunologist  793-904-7725x39308   
1    Tiffany     Myers        Music therapist    201.442.4543x942   
2    Shannon    Watson           Immunologist   687.737.9424x8503   
3     Nathan     Baker  Accounting technician    827-788-8123x012   
4  Christina  Schwartz           Chiropractor        265-829-3643   

                                 email  client_id  order_id  order_date  \
0     harding.donald.7185@sullivan.com      58515   8953482  2023-04-28   
1    myers.t.6537@ferguson-johnson.net      37609   8069089  2023-05-19   
2                swatson8146@payne.net      57113   1902144  2023-01-29   
3               bakernathan@benson.com      46554   9031802  2023-04-25   
4  christinaschwartz9252@mcconnell.com      92089   1322274  2023-05-28   

   order_week  order_year         item_id     category         subcategory  \
0          17        2023  EUD29711-63-6U        decor            wall art   
1          20        2023  XDA18116-89-4A  consumables                pens   
2           4        2023  ABE59463-05-7E     software  project management   
3          17        2023  ZMM00836-65-0C  consumables                pens   
4          21        2023  BZX55559-12-3X  consumables                misc   

   unit_price  unit_cost  unit_weight  qty  line_number   subtotal  
0     1096.80     762.71         7.50  105            1  115164.00  
1       24.95      15.09         1.49   21            0     523.95  
2       13.52       7.86         1.68   39            6     527.28  
3       36.42      24.85         1.23   29            3    1056.18  
4      195.10     108.17        46.43   20            1    3902.00

In [838]:
# Create a column for shipping price.
# Assume a shipping price of $7 per pound
# for orders over 50 pounds and $10 per
# pound for items 50 pounds or under.
def calculate_shipping_cost(weight,qty):
    total_weight = weight*qty
    if total_weight > 50:
        return 7 * total_weight
    else: return 10 * total_weight

updated_df['shipping_price'] = updated_df.apply(lambda row  : calculate_shipping_cost(row['unit_weight'],row['qty']),axis=1)
updated_df.head()


first      last                    job               phone  \
0     Donald   Harding           Immunologist  793-904-7725x39308   
1    Tiffany     Myers        Music therapist    201.442.4543x942   
2    Shannon    Watson           Immunologist   687.737.9424x8503   
3     Nathan     Baker  Accounting technician    827-788-8123x012   
4  Christina  Schwartz           Chiropractor        265-829-3643   

                                 email  client_id  order_id  order_date  \
0     harding.donald.7185@sullivan.com      58515   8953482  2023-04-28   
1    myers.t.6537@ferguson-johnson.net      37609   8069089  2023-05-19   
2                swatson8146@payne.net      57113   1902144  2023-01-29   
3               bakernathan@benson.com      46554   9031802  2023-04-25   
4  christinaschwartz9252@mcconnell.com      92089   1322274  2023-05-28   

   order_week  order_year         item_id     category         subcategory  \
0          17        2023  EUD29711-63-6U        decor            wall art   
1          20        2023  XDA18116-89-4A  consumables                pens   
2           4        2023  ABE59463-05-7E     software  project management   
3          17        2023  ZMM00836-65-0C  consumables                pens   
4          21        2023  BZX55559-12-3X  consumables                misc   

   unit_price  unit_cost  unit_weight  qty  line_number   subtotal  \
0     1096.80     762.71         7.50  105            1  115164.00   
1       24.95      15.09         1.49   21            0     523.95   
2       13.52       7.86         1.68   39            6     527.28   
3       36.42      24.85         1.23   29            3    1056.18   
4      195.10     108.17        46.43   20            1    3902.00   

   shipping_price  
0         5512.50  
1          312.90  
2          458.64  
3          356.70  
4         6500.20

In [839]:
# Create a column for the total price
# using the subtotal and the shipping price
# along with a sales tax of 9.25%
# 125898.61
# 125898.607500
sum = updated_df['subtotal'] + updated_df['shipping_price']
updated_df['total_price'] = (sum + sum * 0.0925).round(2)
updated_df.head()

first      last                    job               phone  \
0     Donald   Harding           Immunologist  793-904-7725x39308   
1    Tiffany     Myers        Music therapist    201.442.4543x942   
2    Shannon    Watson           Immunologist   687.737.9424x8503   
3     Nathan     Baker  Accounting technician    827-788-8123x012   
4  Christina  Schwartz           Chiropractor        265-829-3643   

                                 email  client_id  order_id  order_date  \
0     harding.donald.7185@sullivan.com      58515   8953482  2023-04-28   
1    myers.t.6537@ferguson-johnson.net      37609   8069089  2023-05-19   
2                swatson8146@payne.net      57113   1902144  2023-01-29   
3               bakernathan@benson.com      46554   9031802  2023-04-25   
4  christinaschwartz9252@mcconnell.com      92089   1322274  2023-05-28   

   order_week  order_year  ...     category         subcategory unit_price  \
0          17        2023  ...        decor            wall art    1096.80   
1          20        2023  ...  consumables                pens      24.95   
2           4        2023  ...     software  project management      13.52   
3          17        2023  ...  consumables                pens      36.42   
4          21        2023  ...  consumables                misc     195.10   

   unit_cost  unit_weight  qty  line_number   subtotal  shipping_price  \
0     762.71         7.50  105            1  115164.00         5512.50   
1      15.09         1.49   21            0     523.95          312.90   
2       7.86         1.68   39            6     527.28          458.64   
3      24.85         1.23   29            3    1056.18          356.70   
4     108.17        46.43   20            1    3902.00         6500.20   

   total_price  
0    131839.08  
1       914.26  
2      1077.12  
3      1543.57  
4     11364.40  

[5 rows x 21 columns]

In [840]:
# Create a column for the cost
# of each line using unit cost, qty, and
# shipping price (assume the shipping cost
# is exactly what is charged to the client).
updated_df['cost'] = (updated_df['qty'] * updated_df['unit_cost']) + updated_df['shipping_price']
updated_df.head()


first      last                    job               phone  \
0     Donald   Harding           Immunologist  793-904-7725x39308   
1    Tiffany     Myers        Music therapist    201.442.4543x942   
2    Shannon    Watson           Immunologist   687.737.9424x8503   
3     Nathan     Baker  Accounting technician    827-788-8123x012   
4  Christina  Schwartz           Chiropractor        265-829-3643   

                                 email  client_id  order_id  order_date  \
0     harding.donald.7185@sullivan.com      58515   8953482  2023-04-28   
1    myers.t.6537@ferguson-johnson.net      37609   8069089  2023-05-19   
2                swatson8146@payne.net      57113   1902144  2023-01-29   
3               bakernathan@benson.com      46554   9031802  2023-04-25   
4  christinaschwartz9252@mcconnell.com      92089   1322274  2023-05-28   

   order_week  order_year  ...         subcategory unit_price unit_cost  \
0          17        2023  ...            wall art    1096.80    762.71   
1          20        2023  ...                pens      24.95     15.09   
2           4        2023  ...  project management      13.52      7.86   
3          17        2023  ...                pens      36.42     24.85   
4          21        2023  ...                misc     195.10    108.17   

   unit_weight  qty  line_number   subtotal  shipping_price  total_price  \
0         7.50  105            1  115164.00         5512.50    131839.08   
1         1.49   21            0     523.95          312.90       914.26   
2         1.68   39            6     527.28          458.64      1077.12   
3         1.23   29            3    1056.18          356.70      1543.57   
4        46.43   20            1    3902.00         6500.20     11364.40   

       cost  
0  85597.05  
1    629.79  
2    765.18  
3   1077.35  
4   8663.60  

[5 rows x 22 columns]

In [841]:
# Create a column for the profit of
# each line using line cost and line price
updated_df['profit'] = updated_df['total_price'] - updated_df['cost']
updated_df.head()

first      last                    job               phone  \
0     Donald   Harding           Immunologist  793-904-7725x39308   
1    Tiffany     Myers        Music therapist    201.442.4543x942   
2    Shannon    Watson           Immunologist   687.737.9424x8503   
3     Nathan     Baker  Accounting technician    827-788-8123x012   
4  Christina  Schwartz           Chiropractor        265-829-3643   

                                 email  client_id  order_id  order_date  \
0     harding.donald.7185@sullivan.com      58515   8953482  2023-04-28   
1    myers.t.6537@ferguson-johnson.net      37609   8069089  2023-05-19   
2                swatson8146@payne.net      57113   1902144  2023-01-29   
3               bakernathan@benson.com      46554   9031802  2023-04-25   
4  christinaschwartz9252@mcconnell.com      92089   1322274  2023-05-28   

   order_week  order_year  ... unit_price unit_cost unit_weight  qty  \
0          17        2023  ...    1096.80    762.71        7.50  105   
1          20        2023  ...      24.95     15.09        1.49   21   
2           4        2023  ...      13.52      7.86        1.68   39   
3          17        2023  ...      36.42     24.85        1.23   29   
4          21        2023  ...     195.10    108.17       46.43   20   

   line_number   subtotal  shipping_price  total_price      cost    profit  
0            1  115164.00         5512.50    131839.08  85597.05  46242.03  
1            0     523.95          312.90       914.26    629.79    284.47  
2            6     527.28          458.64      1077.12    765.18    311.94  
3            3    1056.18          356.70      1543.57   1077.35    466.22  
4            1    3902.00         6500.20     11364.40   8663.60   2700.80  

[5 rows x 23 columns]

## Part 3: Confirm your work
You have email receipts showing that the total prices for 3 orders. Confirm that your calculations match the receipts. Remember, each order has multiple lines.

Order ID 2742071 had a total price of \$152,811.89

Order ID 2173913 had a total price of \$162,388.71

Order ID 6128929 had a total price of \$923,441.25


In [842]:
# Check your work using the totals above
# Order 2742071 total: $152811.89
specific_order_1 = 2742071

# Filter the DataFrame based on the specific order ID
filtered_df_1 = updated_df[updated_df['order_id'] == specific_order_1]


selected_columns = filtered_df_1.loc[:, ['order_id', 'total_price']].groupby('order_id')['total_price'].sum()
print(selected_columns)


order_id
2742071    152811.89
Name: total_price, dtype: float64


In [843]:
# Check your work using the totals above
# Order 2173913 total: $162388.71
specific_order_2 = 2173913

# Filter the DataFrame based on the specific order ID
filtered_df_2 = updated_df[updated_df['order_id'] == specific_order_2]

selected_columns_2 = filtered_df_2.loc[:, ['order_id', 'total_price']].groupby('order_id')['total_price'].sum()
print(selected_columns_2)

order_id
2173913    162388.71
Name: total_price, dtype: float64


In [844]:
# Check your work using the totals above
# Order 2173913 total: $923441.25
specific_order_3 = 6128929

# Filter the DataFrame based on the specific order ID
filtered_df_3 = updated_df[updated_df['order_id'] == specific_order_3]

selected_columns_3 = filtered_df_3.loc[:, ['order_id', 'total_price']].groupby('order_id')['total_price'].sum()
print(selected_columns_3)

order_id
6128929    923441.25
Name: total_price, dtype: float64


In [845]:
# Display all 3 reciepts at once
compined_reciept = updated_df.groupby('order_id')['total_price'].sum()
compined_reciept_details = {
    specific_order_1 : format(compined_reciept.loc[specific_order_1], ',.2f') ,
    specific_order_2 : format(compined_reciept.loc[specific_order_2], ',.2f') ,
    specific_order_3 : format(compined_reciept.loc[specific_order_3], ',.2f') 
}
for order,total in compined_reciept_details.items():
    print(f"Order ID {order} had a total price of " + f"${total}")



Order ID 2742071 had a total price of $152,811.89
Order ID 2173913 had a total price of $162,388.71
Order ID 6128929 had a total price of $923,441.25


## Part 4: Summarize and Analyze
Use the new columns with confirmed values to find the following information.

In [846]:
# How much did each of the top 5 clients by quantity
# spend? Check your work from Part 1 for client ids.

total_spending_top_clients = {}

# Calculate total spending for each of the top 5 clients by quantity
for client_id, total_quantity in top_five_clients.items():
    client_df = updated_df[updated_df['client_id'] == client_id]
    total_spending = client_df['total_price'].sum()
    total_spending_top_clients[client_id] = format(total_spending,'.2f')

for client_id, total_spending in total_spending_top_clients.items():
    print(f"{client_id}: $" + f"{total_spending}")



33615: $8377308.52
66037: $10259514.79
46820: $9743794.36
38378: $12906550.87
24741: $82268892.02


In [847]:
# Create a summary DataFrame showing the totals for the
# for the top 5 clients with the following information:
# total units purchased, total shipping price,
# total revenue, and total profit. Sort by total profit.



In [848]:
# Format the data and rename the columns
# to names suitable for presentation.
# Currency should be in millions of dollars.



In [849]:
# Sort the updated data by "Total Profit" form highest to lowest
